fastlane/lib/fastlane/actions/hockey.rb

```diff
@@ -27,24 +27,51 @@ def self.upload(api_token, ipa, options)
         create_update = options.delete(:create_update)
 
         if create_update
-          self.create_and_update_build(api_token, ipa, options)
+          response = self.create_and_update_build(api_token, ipa, options)
         else
-          self.upload_build(api_token, ipa, options)
+          response = self.upload_build(api_token, ipa, options)
+        end
+
+        case response.status
```

Comment:
`create_and_update_build` calls  `upload_ipa_dsym` which already handles the response results and thus doesn't return it.